NEWS.md

```diff
@@ -6,6 +6,11 @@
 
 ## Bug fixes
 
+* DataFrames.jl now checks that passed columns are 1-based as this is a current
+  design assumption; `mapcols!` makes sure not to create columns being
+  `AbstractRange` consistently with other methods that add columns to a `DataFrame`
```

Comment:
This part is probably worth a separate bullet.

Reply:
fixed